src/models/PrimaryAuth.js

```diff
@@ -80,14 +85,26 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
 
       this.appState.trigger('loading', true);
       return this.startTransaction(function (authClient) {
-        return authClient.signIn({
+
+        // Add the custom header for fingerprint if needed, and then remove it afterwards
+        // Since we only need to send it for primary auth
+        if (deviceFingerprintEnabled) {
+          authClient.options.headers['X-Device-Fingerprint'] = this.get('deviceFingerprint');
+        }
+        var signInPromise = authClient.signIn({
           username: username,
           password: password,
           options: {
             warnBeforePasswordExpired: true,
             multiOptionalFactorEnroll: multiOptionalFactorEnroll
           }
+        })
+        .fin(function () {
```

Comment:
Oops, I meant to say "move it to the .fin method" - we're already catching failures & fin in those functions, no need to have it here. Also, just return authClient.signIn instead of creating a variable and returning that.